ui/src/app/base/components/Modal/Modal.js

```diff
@@ -0,0 +1,36 @@
+import PropTypes from "prop-types";
+import React from "react";
+
+export const Modal = ({ children, close, title }) => (
+  <div className="p-modal" onClick={close}>
+    <div
+      className="p-modal__dialog"
+      role="dialog"
+      aria-labelledby="modal-title"
+      aria-describedby="modal-description"
+      onClick={evt => evt.stopPropagation()}
+    >
+      <header className="p-modal__header">
+        <h2 className="p-modal__title" id="modal-title">
```

Comment:
Do you need id's here. That limits the use to one per view.

Reply:
Good point. The ids are used by the aria attributes above, I've updated the component to generate unique ids. I've also filed: https://github.com/canonical-web-and-design/maas-ui/issues/72 to update the form field ids to do the same.